src/Pretzel.Logic/Commands/ICommand.cs

```diff
@@ -0,0 +1,9 @@
+using System.Threading.Tasks;
+
+namespace Pretzel.Logic.Commands
+{
+    public interface IPretzelCommand
```

Comment:
Isn't it possible to link directly the parameters through generic, like:
```cs 
ICommand<T> where T : ICommandParameters
{
    Task Execute(T parameters);
}
```

Another remark, class should have the same name than the file.

Reply:
I thought about `ICommand<in T> where T : ICommandArguments`. Let's see if that is something I can do 👍

Reply:
Have another look, It's not possible to export/import open generics in `System.Composition` this way, but I refactored the whole thing to be more intuitive. I think I should also do that for the ArgumentsExtentions

Reply:
If I do that for `ArgumentExtensions` i will need to export an interface for every arguments class.

Reply:
I don't think so, that seems counterproductive.

Reply:
Hmmm but now after implementing it doesn't look that bad.